pydantic-fastapi-workflow/src/sharing/api/app.py

```diff
@@ -0,0 +1,27 @@
+from fastapi import FastAPI, Request, HTTPException
+from .dependency import get_client #  (untested!)
+# from .queries_merged import whatever you nammed your generated edgeql files
+from .queries_merged import insert_protag_init, InsertProtagInitInput, InsertProtagInitResult
+from fastapi import APIRouter
+
+app = FastAPI()
+
+protagonist_router = APIRouter(prefix="/protagonist")
+
+
+@protagonist_router.put("/update")
+async def insert_protag(data: InsertProtagInitInput) -> InsertProtagInitResult:
+    async with get_client() as client:
```

Comment:
```suggestion
async def insert_protag(data: InsertProtagInitInput, client=Depends(get_client)) -> InsertProtagInitResult:
    async with get_client() as client:
```
Is there a reason why you don't use the dependency injection rather than this additional context manager?

Reply:
no, not at all! i just haven't used DI in fastAPI before, usually in litestar, so have never looked at the fastAPI DI docs. thanks for the heads up! will be using going forwards